app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -242,58 +255,29 @@ def index_organization
     end
 
     api :GET, "/systems/:system_id/subscriptions/available", N_("List available subscriptions"), :deprecated => true
-    param :system_id, String, :desc => N_("UUID of the system"), :required => true
-    param :match_system, :bool, :desc => N_("Return subscriptions that match system")
-    param :match_installed, :bool, :desc => N_("Return subscriptions that match installed")
-    param :no_overlap, :bool, :desc => N_("Return subscriptions that don't overlap")
-    def available_system
+    param :system_id, String, :desc => N_("UUID of the content host"), :required => true
+    param :match_system, :bool, :desc => N_("Return subscriptions that match a content host")
+    param :match_installed, :bool, :desc => N_("Return subscriptions that match installed products")
+    param :no_overlap, :bool, :desc => N_("Return subscriptions which do not overlap with a currently-attached subscription")
+    def available_for_system
       params[:match_system] = ::Foreman::Cast.to_bool(params[:match_system]) if params[:match_system]
       params[:match_installed] = ::Foreman::Cast.to_bool(params[:match_installed]) if params[:match_installed]
       params[:no_overlap] = ::Foreman::Cast.to_bool(params[:no_overlap]) if params[:no_overlap]
       pools = @system.filtered_pools(params[:match_system], params[:match_installed],
                                      params[:no_overlap])
-      available = available_subscriptions(pools, @system.organization)
-
-      subscriptions = {
-        :results => available,
-        :subtotal => available.count,
-        :total => available.count
-      }
+      if pools
+        available = pools.collect { |cp_pool| ::Katello::Pool.find_pool(cp_pool['id']) }
+        available.compact!
+        available.select { |pool| pool.provider?(Organization.find(params[:organization_id])) }
```

Comment:
Having moved a lot of this to the database, this seems like overkill to find a set of available pools.